### ISSUE TITLE
remove "regulation of  catalytic activity"  from  BP

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-ontology#12995]
protein kinase activator activity
Ontology	Molecular Function
Definition	Binds to and increases the activity of a protein kinase, an enzyme which phosphorylates a protein.

is a descendent of
GO:0032147 activation of protein kinase activity
Any process that initiates the activity of an inactive protein kinase.

The problem is that the MF branch does not absolutely specify 
that an enzyme activator activity 
absolutely  initiates the activity of an inactive *enzyme*

I think it should?
Otherwise its open to misinterpretation.....

**Comment by dosumis** [comment]
I think we should obsolete all of the process terms 'activation of fu activity' (BP).  It's too confusing to have these and the 'fu activity activation' terms (MF).

**Comment by dosumis** [comment]
> The problem is that the MF branch does not absolutely specify
> that an enzyme activator activity
> absolutely initiates the activity of an inactive enzyme

> I think it should?
> Otherwise its open to misinterpretation.....


In discussion in LA, it was generally agreed that we shouldn't distinguish direct activation from direct positive regulation.  I agree that this makes the name 'activator' a little tricky.  

In logical definitions, 'X activator activity' is (now) defined as any MF that directly_positively_regulates some X.    We need to update all the text definitions to fit this.  How about:

protein kinase activator activity
Direct interaction with a gene product or complex that has protein kinase activity, resulting in positive regulation of its protein kinase activity.  Direct interaction in this case includes binding and interactions leading to covalent modification.

**Comment by dosumis** [comment]
[Draft template](https://github.com/geneontology/design_patterns/blob/master/validated_patterns/activator_activity.yaml)

**Comment by ValWood** [comment]
Hi David,

I think that your solution makes sense based on the decision. I am still slightly  nervous that this decision will not allow us to capture biology precisely, but I think the more general approach is probably sensible here at this stage....

v

**Comment by tberardini** [comment]
>  We need to update all the text definitions to fit this. 

@dosumis : Computationally? :)

**Comment by dosumis** [comment]
> >  We need to update all the text definitions to fit this. 

> @dosumis : Computationally? :)

Ideally. This will require specification of the pattern fillers (regulated activity for each class).  We have these for a minority of the 203 classes (those already with logical defs).  The rest would need to be specified by hand.  

@balhoff How close is your DOSDP scowl library to being usable for a retrofit like this? For those already with logical defs, we could use your SPARQL query system to pull out fillers.  For the rest, I could potentially provide a spreadsheet of IDs and fillers.  For the retrofit, we'd then need to be able to change existing textual defs, ideally preserving any existing refs.  Given that, perhaps the easiest approach would for me to extract defs separately into the spreadsheet specifying IDs and fillers (easy enough to do from obo-graphs JSON).

CC @cmungall

**Comment by dosumis** [comment]
Update:  I've generated a [tsv file](https://gist.github.com/dosumis/4d899727896367c9b432e733e120b675)  => current spec for this branch using OBO-graphs.   This includes definition xrefs + pattern filler column. About 40 have logical defs so far, a few of which don't conform to the standard pattern.  The rest need mapping to the regulated function. 

@balhoff: With that in place + pattern docs, could we use scowl to update (& ultimately automate) the branch?

**Comment by dosumis** [comment]
> I think we should obsolete all of the process terms 'activation of fu activity' (BP). It's too confusing to have these and the 'fu activity activation' terms (MF).

activation of fu activity terms - potential candidates for obsoletion as single step BPs (involved_in ones should be looked at carefully - check if direct?)

'activation of GTPase activity'
'activation of JAK1 kinase activity'
'activation of JAK2 kinase activity'
'activation of JNKK activity'
'activation of JNKKK activity'
'activation of JUN kinase activity'
'activation of Janus kinase activity'
'activation of MAPK activity by adrenergic receptor signaling pathway'
'activation of MAPK activity involved in cell wall organization or biogenesis'
'activation of MAPK activity involved in conjugation with cellular fusion'
'activation of MAPK activity involved in innate immune response'
'activation of MAPK activity involved in osmosensory signaling pathway'
'activation of MAPK activity'
'activation of MAPKK activity involved in cell wall organization or biogenesis'
'activation of MAPKK activity involved in conjugation with cellular fusion'
'activation of MAPKK activity involved in innate immune response'
'activation of MAPKK activity involved in osmosensory signaling pathway'
'activation of MAPKK activity'
'activation of MAPKKK activity involved in cell wall organization or biogenesis'
'activation of MAPKKK activity involved in conjugation with cellular fusion'
'activation of MAPKKK activity involved in innate immune response'
'activation of MAPKKK activity involved in osmosensory signaling pathway'
'activation of MAPKKK activity'
'activation of NF-kappaB-inducing kinase activity'
'activation of adenylate cyclase activity'
'activation of cysteine-type endopeptidase activity involved in apoptotic process by cytochrome c'
'activation of cysteine-type endopeptidase activity involved in apoptotic process'
'activation of cysteine-type endopeptidase activity involved in apoptotic signaling pathway'
'activation of cysteine-type endopeptidase activity involved in execution phase of apoptosis'
'activation of cysteine-type endopeptidase activity'
'activation of phospholipase A2 activity by calcium-mediated signaling'
'activation of phospholipase A2 activity'
'activation of phospholipase C activity'
'activation of phospholipase D activity'
'activation of protein histidine kinase activity'
'activation of protein kinase A activity'
'activation of protein kinase B activity'
'activation of protein kinase C activity'
'activation of protein kinase activity'
'activation of signaling protein activity involved in unfolded protein response'
'activation of store-operated calcium channel activity'
'activation of transmembrane receptor protein tyrosine kinase activity'
'activation of tryptophan 5-monooxygenase activity'
'inactivation of MAPK activity involved in cell wall organization or biogenesis'
'inactivation of MAPK activity involved in conjugation with cellular fusion'
'inactivation of MAPK activity involved in innate immune response'
'inactivation of MAPK activity involved in osmosensory signaling pathway'
'inactivation of MAPK activity'
'inactivation of MAPKK activity'
'inactivation of MAPKKK activity'
'negative regulation of activation of JAK2 kinase activity'
'positive regulation of activation of JAK1 kinase activity'
'positive regulation of activation of JAK2 kinase activity'
'positive regulation of activation of Janus kinase activity'
'regulation of activation of JAK1 kinase activity'
'regulation of activation of JAK2 kinase activity'
'regulation of activation of Janus kinase activity'

**Comment by ValWood** [comment]
good plan ;)

or merge up?

**Comment by dosumis** [comment]
update: I added some name parsing to the tsv mapping script.  The [tsv mapping file](https://gist.github.com/dosumis/4d899727896367c9b432e733e120b675) for MF regulator activity terms now has 146 classes with a mapping.  52 that need manually mapping.

**Comment by dosumis** [comment]
> good plan ;)
> or merge up?

Merge down to corresponding MF term?  One worry is whether the process anotations will include some indirect.

**Comment by ValWood** [comment]
I don't think you can merge down. Generally if the MF "enzyme regulator" term is valid,  the curator should have made the MF annotation. We tend to  use the process term in direct annotations if it is definitely not, or not known to be, acting directly on the enzyme.

Also at PomBAse don't usually annotate kinase regulators which are kinases to these terms:

MAPKK activator activity 
for MAPKKK.
We would use "MAPKKKK activity" +ve_regulates_activity_of "MapKK"

Otherwise we have 2 ways of describing the same thing:
"MAPKKKK activity" +ve_regulates_activity_of "specific MAPKK"
and
MAPKK activator activity  +ve_regulates_activity_of "specific MAPKK"

we decided to be consistent, and use the MF term that describes the actual activity.

(I would link this to my slide about this at GO meeting if I know where to find it....later will do)
(OK, this is a PomBase specific reason).

I don't think its good practice to merge BP terms into MF terms, they are used slightly differently...

**Comment by dosumis** [comment]
The MF terms will stay.  They're part of the MF refactoring.  The pattern and tsv linked to above are for this.  

The inference is such that if you annotate to this:

"MAPKKK activity"  directly_positively_regulates  "MapKK activity"

You'll get an inferred annotation to 'MAPKK activator activity'.


If the BP terms are used widely for indirect activation, then maybe we should keep them.  But it does rather make my head spin that we have both 'protein kinase activator activity' and 'activation of protein kinase activity'.  We do, in many cases have a superclass in BP: 'positive regulation of protein kinase activity'.  I think the solution is to merge 'activation of ...' up into 'positive regulation of ...'.

**Comment by dosumis** [comment]
- [ ] TODO: Use design pattern + tsv mapping file to update MF regulator activity terms in ontology.  Moved to #13492
- [ ] TODO: Merge all 'activation/inhibition of fu' activity terms up to their parent 'negative/positive regulation of fu activity parent'  (Can be done independently of #13492)

**Comment by ValWood** [comment]
> I think the solution is to merge 'activation of ...' up into 'positive regulation of ...'.

I agree....but probably needs a heads up from other groups. I'm happy for the change but I was only reporting the inconsistency.  We would want to reannotate, or at least check, if a merge up occurs...

**Comment by pgaudet** [comment]
At the March 2nd Editor's call we agreed with @dosumis 's plan to merge activation and positive regulation in Biological processes. For direct activation, there is 'activator activity' MF that can be used.

**Comment by tberardini** [comment]
Discussed on March 2 meeting.  Editors agree with DOS' plan to "I think we should obsolete [modified to merge up into pos/neg reg terms] all of the process terms 'activation [inhibition} of fu activity' (BP). It's too confusing to have these and the 'fu activity activator [inhibitor]' terms (MF)."

**Comment by RLovering** [comment]
I don't know how many annotations are associated with the terms you plan to obsolete, however as it was only recently agreed that in order to apply the MF term 'activator' or 'inhibitor' direct binding has to be shown and I know that I revised UCL annotations following this (although I think that many revised were highly likely to be 'activators or inhibitors' but the direct binding for the protein in question wasn't available due to lack of species information for example). 
Anyway I think it would be safer to move these annotations to the pos/neg reg terms, rather than changing to the MF  'activator' or 'inhibitor' terms. I am sure annotation curators would appreciate being sent a list of proteins which have been revised in this way so that they can decide if they want to edit these. 
I guess it would be useful to have the list of proteins that are also annotated to the MF term then curators could decide whether they want to check if the proteins not annotated to the MF term should be annotated to this term.

Ruth

**Comment by tberardini** [comment]
> "I think we should obsolete [modified to merge up into pos/neg reg terms] all of the process terms 'activation [inhibition} of fu activity' (BP). 

The "**obsolete [modified to merge up into pos/neg reg terms]**" means that the agreement was to MERGE the terms into pos/neg reg terms NOT OBSOLETE them.

**Comment by dosumis** [comment]
> Anyway I think it would be safer to move these annotations to the pos/neg reg terms, rather than changing to the MF 'activator' or 'inhibitor' terms.

The MF terms will remain and will be defined as direct.  The corresponding BP terms will be merged up.   (It's confusing I know, but that's why we're need to change).

>  however as it was only recently agreed that in order to apply the MF term 'activator' or 'inhibitor' direct binding has to be shown 

Strictly - direct interaction.  So doesn't have to be stable binding e.g.-  Could be a kinase regulating its substrate.  This is built into the definition of the direct reg relations (recently updated to make clearer), which are used in logical defs of these terms and are widely used in LEGO.

**Comment by pgaudet** [comment]
Sounds good !

**Comment by ValWood** [comment]
@pgaudet  I had forgotten there was a ticket for this. Is there a duplicate anywhere. The suggestion here is that it can be done computationally.

**Comment by ValWood** [comment]
regulation of catalytic activity still has 4001 EXP annotations:
(466,100 total all evidence codes in QuickGO)

<img width="622" alt="Screenshot 2022-11-19 at 11 03 50" src="https://user-images.githubusercontent.com/7359272/202847581-2bc02416-9ee1-4f3d-914a-3959da3b5b85.png">

with a really long tail, tonnes of terms with a single annotation. This would be too much for ontology editors to review. Would it be possible to obsolete the entire set and let individual  resources  either review, or not  as they choose ( they should most likely be correctly annotated to the corresponding MF term
 "GTPase activator" "protein kinase activator". etc, unless they were wrong, in which case nothing is lost.

**Comment by hattrill** [comment]
Hi @ValWood I would be fine with this - needs to be done. Perhaps, pick a month in the new year and do it (pre-Xmas might be too much as so many deadlines).

**Comment by RLovering** [comment]
if you could create a list for me to review I would really appreciate it - Ruth

**Comment by ValWood** [comment]
@pgaudet  how can we get the complete term list?

**Comment by ValWood** [comment]
@pgaudet  is there another ticket for this somewhere?
I'm unassigning myself as this should be automated in some way?

**Comment by ValWood** [comment]
Can we close this. I presume this is covered by other tickets/in progress (they all have "This term should not be used for direct annotation" but should it be "obsoletion candidate"?

Anyway the number of EXP is dropping. now down to 1,956 annotations

**Comment by pgaudet** [comment]
Sure !!

**Comment by ValWood** [comment]
down to 1412 EXP

**Comment by ValWood** [comment]
1429, how is this going UP when the terms are "do not annotate"?????

**Comment by ValWood** [comment]
out of data anyway, duplicate of https://github.com/geneontology/go-ontology/issues/22455
I'll keep the newer one
